src/commands/dump.rs

```diff
@@ -24,29 +24,39 @@ pub struct Guard {
 
 
 impl Guard {
-    async fn open(filename: &Path) -> anyhow::Result<(Output, Guard)> {
+    async fn open(filename: &Path, overwrite_existing: bool) -> anyhow::Result<(Output, Guard)> {
+
         if filename.to_str() == Some("-") {
-            Ok((Box::new(io::stdout()), Guard { filenames: None }))
-        } else if cfg!(windows)
-            || filename.starts_with("/dev/")
-            || filename.file_name().is_none()
+            return Ok((Box::new(io::stdout()), Guard { filenames: None }))
+        }
+
+        let mut options = OpenOptions::new();
+        let options = match overwrite_existing {
+            true => options.write(true).create(true).truncate(true),
+            false => options.write(true).create_new(true)
+        };
+
+        if cfg!(windows) || filename.starts_with("/dev/") || filename.file_name().is_none()
         {
-            let file = fs::File::create(&filename).await
+            let file = options
+                .open(&filename).await
                 .context(filename.display().to_string())?;
             Ok((Box::new(file), Guard { filenames: None }))
         } else {
             let tmp_path = filename.with_file_name(
                 tmp_file_name(filename.as_ref()));
-            if fs::metadata(&tmp_path).await.is_ok() {
+            if fs::metadata(&tmp_path).await.is_ok() && overwrite_existing {
```

Comment:
I think this doesn't seem right - the temporary file doesn't usually exist, and we should instead protect the `fs::rename()` in `fn commit()`.